tests/python_client/testcases/test_collection.py

```diff
@@ -2161,6 +2162,254 @@ def test_load_partitions_release_collection(self):
         patition_w.load()
         collection_w.release()
 
+    @pytest.fixture(scope="function", params=ct.get_invalid_strs)
+    def get_non_number_replicas(self, request):
+        if request.param == 1:
+            pytest.skip("1 is valid replica number")
+        if request.param is None:
+            pytest.skip("None is valid replica number")
+        yield request.param
+
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_load_replica_non_number(self, get_non_number_replicas):
+        """
+        target: test load collection with non-number replicas
+        method: load with non-number replicas
+        expected: raise exceptions
+        """
+        # create, insert
+        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
+        df = cf.gen_default_dataframe_data()
+        insert_res, _ = collection_w.insert(df)
+        assert collection_w.num_entities == ct.default_nb
+
+        # load with non-number replicas
+        error = {ct.err_code: 0, ct.err_msg: f"but expected one of: int, long"}
+        collection_w.load(replica_number=get_non_number_replicas, check_task=CheckTasks.err_res, check_items=error)
+
+    @pytest.mark.tags(CaseLabel.L2)
+    @pytest.mark.parametrize("replicas", [-1, 0, None])
+    def test_load_replica_invalid_number(self, replicas):
+        """
+        target: test load partition with invalid replica number
+        method: load with invalid replica number
+        expected: raise exception
+        """
+        # create, insert
+        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix))
+        df = cf.gen_default_dataframe_data()
+        insert_res, _ = collection_w.insert(df)
+        assert collection_w.num_entities == ct.default_nb
+
+        collection_w.load(replica_number=replicas)
```

Comment:
no errors here for invalid replicas?

Reply:
No exception is thrown by design. This is for compatibility with older versions without the replica_number field, and the value obtained from the request is 0. So if the replica_number is None or less than 0, milvus will set it to 1.